### PR TITLE
Added the `@dbsp.error.abort()` function and the `unreachable` terminator

### PR DESCRIPTION
**Pull request by Kixiron**
Adds the `@dbsp.error.abort()` function to immediately abort the program and the `unreachable` terminator for terminating blocks with no meaningful branching/return behavior, currently should just be used as
```llvm
call @dbsp.error.abort()
unreachable
```

### CODE REVIEW

File: crates/dataflow-jit/src/codegen/call.rs [hunk 1]
```diff
@@ -100,6 +103,16 @@ impl CodegenCtx<'_> {
         }
     }
 
+    fn error_abort(&self, builder: &mut FunctionBuilder<'_>) {
+        let trap = builder.ins().trap(TRAP_ABORT);
+
+        if let Some(writer) = self.comment_writer.as_deref() {
+            writer
+                .borrow_mut()
+                .add_comment(trap, "call @dbsp.error.abort()");
```
Comment (mbudiu-vmw) [inline crates/dataflow-jit/src/codegen/call.rs:112]: in the future we should add information to help users identify where the error happened. For example, having some kind of error message from the compiler could work.

**Review by mbudiu-vmw — APPROVED** [review]
How about adding some JSON-based tests too which show how these should be used by the compiler?

File: crates/dataflow-jit/src/codegen/call.rs (hunk 1 above)
Comment (Kixiron) [reply crates/dataflow-jit/src/codegen/call.rs:112]: Yah, I'll get around to that

**Comment by Kixiron** [comment]
Fixes #368, #371 and #369

**Pull request merged**